pkg/providers/instance/instance.go

```diff
@@ -353,6 +356,12 @@ func (p *Provider) updateUnavailableOfferingsCache(ctx context.Context, errors [
 	for _, err := range errors {
 		if awserrors.IsUnfulfillableCapacity(err) {
 			p.unavailableOfferings.MarkUnavailableForFleetErr(ctx, err, capacityType)
+			// Add a k8s event for the instance type and zone without the involved object which has an ICE error
+			p.recorder.Publish(events.Event{
+				Type:    v1.EventTypeWarning,
+				Reason:  "InsufficientCapacityError",
+				Message: fmt.Sprintf("got InsufficientCapacityError for the instanceType %s, availabilityZone %s, capacityType %s", aws.StringValue(err.LaunchTemplateAndOverrides.Overrides.InstanceType), aws.StringValue(err.LaunchTemplateAndOverrides.Overrides.AvailabilityZone), capacityType),
```

Comment:
I think we also need a dedupe value here as well since this can fire off quite a few times if CreateFleet requests are batched together. 

```suggestion
			// Add a k8s event for the instance type and zone without the involved object which has an ICE error
			instanceType := aws.StringValue(err.LaunchTemplateAndOverrides.Overrides.InstanceType)
			availabilityZone := aws.StringValue(err.LaunchTemplateAndOverrides.Overrides.AvailabilityZone)
			p.recorder.Publish(events.Event{
				Type:    v1.EventTypeWarning,
				Reason:  "InsufficientCapacityError",
				Message: fmt.Sprintf("got InsufficientCapacityError for the instanceType %s, availabilityZone %s, capacityType %s", instanceType, availabilityZone, capacityType),
				DedupeValues: []string{fmt.Sprintf("ice-%s-%s-%s", instanceType, availabilityZone, capacityType)},
```